Support multiple case references per form question

millerdev opened the pull request:
Next up: make vellum send case references in the new format, which supports multiple case properties per question. This PR is backward-compatible with the old format, which is currently used in vellum.

@emord cc @esoergel

millerdev committed:
Support multiple case references per form question

emord commented on corehq/apps/app_manager/models.py:
```diff
@@ -1305,6 +1305,7 @@ class Form(IndexedFormBase, NavMenuItemMediaMixin):
     form_filter = StringProperty()
     requires = StringProperty(choices=["case", "referral", "none"], default="none")
     actions = SchemaProperty(FormActions)
+    case_references_data = DictProperty()
```
I like the idea of having a generic case references property, but I think it could cause some confusion when saving is implemented as this is actually specific to loading

millerdev replied:
Good point. I'll change the data structure to support both loads and saves.

millerdev replied:
3cec08d

millerdev committed:
Add "load" item to Form.case_references

A "save" item with a similar format to the "load" item will be added to the format when the form builder gets support for save to case.

emord commented:
:+1: on :white_check_mark:

millerdev commented:
✅
